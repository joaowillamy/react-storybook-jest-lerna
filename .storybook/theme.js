import { create } from "@storybook/theming/create";

export default create({
  base: "light",

  colorPrimary: "#FDCB13",
  colorSecondary: "#18ACC4",

  // UI
  appBg: "white",
  appContentBg: "#F2F3F4",
  appBorderColor: "#F2F3F4",
  appBorderRadius: 4,

  // Typography
  fontBase: '"Montserrat", sans-serif',
  fontCode: "monospace",

  brandTitle: "Quero front test"
  // brandUrl: 'http://resolvvi.com.br/',
  // brandImage: ''
});
