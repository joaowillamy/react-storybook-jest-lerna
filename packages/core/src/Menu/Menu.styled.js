import styled from "styled-components";
import media from "styled-media-query";

import { chooseScrem } from "../help/chooseScreem";

export const Menu = styled.div`
  background-color: ${({ theme }) => theme.colors.bluePrimary};
`;

export const MenuItems = styled.ul`
  ${media.lessThan("medium")`
    display: flex;
    justify-content: space-between;
    align-items: center;
  `}
`;

export const MenuItem = styled.li`
  display: inline-block;
  padding: 15px;
  font-weight: 600;
  color: #fff;
  ${({ isActive, theme }) =>
    isActive && `background-color: ${theme.colors.blueSecondary}`}
  ${chooseScrem};
  transition: all 0.3s;
  cursor: pointer;

  :hover {
    background-color: ${({ theme }) => theme.colors.blueSecondary};
  }

  svg {
    width: ${({ theme }) => theme.iconSizes.lg};
    color: #fff;
  }
`;
