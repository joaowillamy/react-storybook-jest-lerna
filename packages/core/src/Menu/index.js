import React from "react";
import PropTypes from "prop-types";
import { Icon } from "../Icons";

import { Container } from "../help/Container.styled";
import * as S from "./Menu.styled";

const Menu = props => (
  <S.Menu {...props}>
    <nav>
      <Container>
        <S.MenuItems>
          <S.MenuItem>Minha conta</S.MenuItem>
          <S.MenuItem justDesktop>Pré-matrículas</S.MenuItem>
          <S.MenuItem isActive justDesktop>
            Bolsas favoritas
          </S.MenuItem>
          <S.MenuItem justMobile>
            Menu <Icon.ArrowDownward />
          </S.MenuItem>
        </S.MenuItems>
      </Container>
    </nav>
  </S.Menu>
);

Menu.propTypes = {
  options: PropTypes.array
};

export default Menu;
