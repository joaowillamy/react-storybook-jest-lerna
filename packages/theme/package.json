{
  "name": "@will-ui-kit/theme",
  "version": "1.0.2",
  "main": "lib/index.js",
  "module": "src/index.js"
}
