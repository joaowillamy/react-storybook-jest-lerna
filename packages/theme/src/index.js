export { default as ThemeProvider } from "./ThemeProvider";
export { default as DEFAULT_THEME } from "./theme";
export { GlobalStyles } from "./GlobalStyles";
