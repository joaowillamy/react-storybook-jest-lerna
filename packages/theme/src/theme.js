const BASE = 3;

const borders = {
  sm: color => `1px solid ${color || "#00000029"}`,
  md: color => `3px solid ${color || "#00000029"}`
};

const colors = {
  bluePrimary: "#18ACC4",
  blueSecondary: "#007A8D",
  yellowPrimary: "#FDCB13",
  yellowSecondary: "#DE9E1F",
  green: "#0FA866",
  black: "#1F2D30",
  grey: "#FBFBFB",
  Overlay: "#1F2D30e0" // com 88% de opacidade.
};

const fonts = {
  Montserrat: "Montserrat"
};

const fontSizes = {
  xs: "12px",
  sm: "14px",
  md: "16px",
  lg: "18px",
  xl: "22px",
  xxl: "26px",
  xxxl: "36px"
};

const fontWeights = {
  thin: 100,
  extralight: 200,
  light: 300,
  regular: 400,
  medium: 500,
  semibold: 600,
  bold: 700,
  extrabold: 800,
  black: 900
};

const iconSizes = {
  sm: "12px",
  md: "16px",
  lg: "26px"
};

const shadows = {
  sm: color => `0 1px 3px ${color || "#00000029"}`,
  md: color => `0 3px 4px ${color || "#00000029"}`,
  lg: color => `0 3px 6px ${color || "#00000029"}`
};

const space = {
  xs: `${BASE}px`,
  sm: `${BASE * 2}px`,
  md: `${BASE * 3}px`,
  lg: `${BASE * 4}px`,
  xl: `${BASE * 8}px`,
  xxl: `${BASE * 10}px`,
  xxxl: `${BASE * 12}px`
};

const DEFAULT_THEME = {
  borders,
  colors,
  fonts,
  fontSizes,
  fontWeights,
  iconSizes,
  shadows,
  space
};

export default DEFAULT_THEME;
