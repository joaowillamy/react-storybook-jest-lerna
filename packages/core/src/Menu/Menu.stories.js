import React from "react";
import { ThemeProvider } from "@joaowillamy-test-quero/theme";

import Menu from ".";

export default {
  component: Menu,
  title: "Core|Menu"
};

export const menu = () => (
  <ThemeProvider>
    <Menu />
  </ThemeProvider>
);
