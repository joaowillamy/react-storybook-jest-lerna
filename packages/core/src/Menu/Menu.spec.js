import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";
import { ThemeProvider } from "@joaowillamy-test-quero/theme";

import Menu from ".";

describe("Menu", () => {
  test("Menu", () => {
    const tree = renderer
      .create(
        <ThemeProvider>
          <Menu />
        </ThemeProvider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
