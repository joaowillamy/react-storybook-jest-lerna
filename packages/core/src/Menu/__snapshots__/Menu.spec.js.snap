// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Menu 1`] = `
.c7 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c0 {
  background-color: #18ACC4;
}

.c3 {
  display: inline-block;
  padding: 15px;
  font-weight: 600;
  color: #fff;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  cursor: pointer;
}

.c3:hover {
  background-color: #007A8D;
}

.c3 svg {
  width: 26px;
  color: #fff;
}

.c4 {
  display: inline-block;
  padding: 15px;
  font-weight: 600;
  color: #fff;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  cursor: pointer;
}

.c4:hover {
  background-color: #007A8D;
}

.c4 svg {
  width: 26px;
  color: #fff;
}

.c5 {
  display: inline-block;
  padding: 15px;
  font-weight: 600;
  color: #fff;
  background-color: #007A8D;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  cursor: pointer;
}

.c5:hover {
  background-color: #007A8D;
}

.c5 svg {
  width: 26px;
  color: #fff;
}

.c6 {
  display: inline-block;
  padding: 15px;
  font-weight: 600;
  color: #fff;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  cursor: pointer;
}

.c6:hover {
  background-color: #007A8D;
}

.c6 svg {
  width: 26px;
  color: #fff;
}

@media (min-width:768px) {
  .c1 {
    width: 95%;
    margin: 0 auto;
  }
}

@media (max-width:768px) {
  .c2 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

@media (max-width:768px) {
  .c4 {
    display: none;
  }
}

@media (max-width:768px) {
  .c5 {
    display: none;
  }
}

@media (min-width:768px) {
  .c6 {
    display: none;
  }
}

<div
  className="c0"
>
  <nav>
    <div
      className="c1"
    >
      <ul
        className="c2"
      >
        <li
          className="c3"
        >
          Minha conta
        </li>
        <li
          className="c4"
        >
          Pré-matrículas
        </li>
        <li
          className="c5"
        >
          Bolsas favoritas
        </li>
        <li
          className="c6"
        >
          Menu 
          <svg
            aria-hidden="true"
            className="c7"
            fill="currentColor"
            focusable="false"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              data-name="Layer 2"
            >
              <path
                d="M12 16a1 1 0 01-.64-.23l-6-5a1 1 0 111.28-1.54L12 13.71l5.36-4.32a1 1 0 011.41.15 1 1 0 01-.14 1.46l-6 4.83A1 1 0 0112 16z"
                data-name="arrow-ios-downward"
              />
            </g>
          </svg>
        </li>
      </ul>
    </div>
  </nav>
</div>
`;
